File: src/main/scala/com/katus/chapter10/IteratorTest.scala
package com.katus.chapter10

/**
 * @author SUN Katus
 * @version 1.0, 2020-08-20
 */
object IteratorTest {
  def main(args: Array[String]): Unit = {
    test1()
  }

  def test1(): Unit = {
    val lst = List(1, 2, 3, 4, 5, 6)
    val it = lst.iterator
    while (it.hasNext) {
      println(it.next())
    }
  }
}

File: src/main/java/com/katus/collections/IteratorTest.java
package com.katus.collections;

import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;
import java.util.Iterator;

/**
 * @author SUN Katus
 * @version 1.0, 2020-08-15
 */
public class IteratorTest {
    @Test
    public void test1() {
        Collection coll = new ArrayList<>();
        coll.add(123);
        coll.add("456");
        coll.add(new Date());

        Iterator it = coll.iterator();   // 每次调用都会返回一个新的迭代器
        while (it.hasNext()) {
            Object obj = it.next();
            if (obj.equals("456")) {
                // 只允许在每次调用next方法之后调用一次
                it.remove();   // 删除集合中的元素
            }
        }

        System.out.println(coll);

        for (Object obj : coll) {
            System.out.println(obj);
        }
    }
}
